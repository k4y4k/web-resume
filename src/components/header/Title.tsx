import * as React from 'react'

interface TitleProps {
  title: string
}

const Title = ({ title }: TitleProps): JSX.Element | null => {
  if (title === '') return null

  return <h1 data-testid='headerTitle'>{title}</h1>
}

export default Title
