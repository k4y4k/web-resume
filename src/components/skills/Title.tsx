import * as React from 'react'
import tw, { css } from 'twin.macro'

interface TitleTypes {
  category: string
}

const TitleStyles = css`
  ${tw`text-2xl font-black uppercase`}
  ${tw`print:(text-lg)`}
`

const Title = ({ category }: TitleTypes): JSX.Element => {
  if (category === '')
    return <h2 css={[TitleStyles]}>Error: No Skills Category</h2>

  return <h2 css={[TitleStyles]}>{category}</h2>
}
export default Title
