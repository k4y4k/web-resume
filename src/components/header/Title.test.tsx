import * as React from "react";
import { render, screen } from "@testing-library/react";
import Title from "./Title";

describe("<Title />", () => {
  test("Handles not having any data properly", () => {
    render(<Title lightmode={false} compact={false} title="" />);

    const title = screen.queryByTestId("headerTitle");
    expect(title).not.toBeInTheDocument();
    expect(title).toMatchSnapshot();
  });

  test("displays title", () => {
    render(<Title lightmode={false} compact={false} title="kayak kayak" />);

    const title = screen.getByTestId("headerTitle");
    expect(title).toHaveTextContent("kayak kayak");
    expect(title).toMatchSnapshot();
  });

  test("lightmode", () => {
    render(<Title lightmode={true} compact={false} title="kayak kayak" />);

    const title = screen.getByTestId("headerTitle");
    expect(title).toHaveTextContent("kayak kayak");
    expect(title).toMatchSnapshot();
  });

  test("compact", () => {
    render(<Title lightmode={false} compact={true} title="kayak kayak" />);

    const title = screen.getByTestId("headerTitle");
    expect(title).toHaveTextContent("kayak kayak");
    expect(title).toMatchSnapshot();
  });

  test("lightmode and compact", () => {
    render(<Title lightmode={true} compact={true} title="kayak kayak" />);

    const title = screen.getByTestId("headerTitle");
    expect(title).toHaveTextContent("kayak kayak");
    expect(title).toMatchSnapshot();
  });
});
