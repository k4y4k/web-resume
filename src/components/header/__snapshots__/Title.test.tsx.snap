// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> Handles not having any data properly 1`] = `null`;

exports[`<Title /> displays title 1`] = `
.c0 {
  font-weight: 700;
  font-family: Inconsolata,ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,Liberation Mono,Courier New,monospace;
  font-size: 3rem;
  line-height: 1;
  margin-bottom: 0.5rem;
}

<h1
  class="c0"
  data-testid="headerTitle"
>
  kayak kayak
</h1>
`;
